commit 84f9a9628f63b2868f7daf07ed431bc51b562c39
Author: Ukang'a Dickson <ukanga@users.noreply.github.com>
Date:   Fri Jan 21 06:22:18 2022 +0300

    [PATCH] status color

diff --git a/config.def.h b/config.def.h
index 70dfb55..e4a7d9f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -18,17 +18,27 @@ static char normfgcolor[]           = "#bbbbbb";
 static char selfgcolor[]            = "#eeeeee";
 static char selbordercolor[]        = "#005577";
 static char selbgcolor[]            = "#005577";
-static const unsigned int baralpha = 0xd0;
-static const unsigned int borderalpha = OPAQUE;
+static char warnbordercolor[]       = "#ff0000";
+static char warnbgcolor[]           = "#ffff00";
+static char warnfgcolor[]           = "#000000";
+static char urgentfgcolor[]         = "#ffffff";
+static char urgentbordercolor[]     = "#ff0000";
+static char urgentbgcolor[]         = "#ff0000";
 static char *colors[][3] = {
        /*               fg           bg           border   */
-       [SchemeNorm] = { normfgcolor, normbgcolor, normbordercolor },
-       [SchemeSel]  = { selfgcolor,  selbgcolor,  selbordercolor  },
+       [SchemeNorm]    = { normfgcolor,    normbgcolor,    normbordercolor },
+       [SchemeSel]     = { selfgcolor,     selbgcolor,     selbordercolor  },
+       [SchemeWarn]    = { warnfgcolor,    warnbgcolor,    warnbordercolor  },
+       [SchemeUrgent]  = { urgentfgcolor,  urgentbgcolor,  urgentbordercolor  },
 };
+static const unsigned int baralpha = 0xd0;
+static const unsigned int borderalpha = OPAQUE;
 static const unsigned int alphas[][3]      = {
 	/*               fg      bg        border     */
-	[SchemeNorm] = { OPAQUE, baralpha, borderalpha },
-	[SchemeSel]  = { OPAQUE, baralpha, borderalpha },
+	[SchemeNorm]    = { OPAQUE, baralpha, borderalpha },
+	[SchemeSel]     = { OPAQUE, baralpha, borderalpha },
+	[SchemeWarn]    = { OPAQUE, baralpha, borderalpha },
+	[SchemeUrgent]  = { OPAQUE, baralpha, borderalpha },
 };
 
 /* tagging */
diff --git a/dwm.c b/dwm.c
index fa22e40..01617e3 100644
--- a/dwm.c
+++ b/dwm.c
@@ -77,7 +77,7 @@
 
 /* enums */
 enum { CurNormal, CurResize, CurMove, CurLast }; /* cursor */
-enum { SchemeNorm, SchemeSel }; /* color schemes */
+enum { SchemeNorm, SchemeSel, SchemeWarn, SchemeUrgent }; /* color schemes */
 enum { NetSupported, NetWMName, NetWMState, NetWMCheck,
        NetWMFullscreen, NetActiveWindow, NetWMWindowType,
        NetWMWindowTypeDialog, NetClientList, NetLast }; /* EWMH atoms */
@@ -842,6 +842,10 @@ drawbar(Monitor *m)
 	int boxs = drw->fonts->h / 9;
 	int boxw = drw->fonts->h / 6 + 2;
 	unsigned int i, occ = 0, urg = 0;
+ 	char *ts = stext;
+ 	char *tp = stext;
+	int tx = 0;
+	char ctmp;
 	Client *c;
 
 	if (!m->showbar)
@@ -851,7 +855,17 @@ drawbar(Monitor *m)
 	if (m == selmon) { /* status is only drawn on selected monitor */
 		drw_setscheme(drw, scheme[SchemeNorm]);
 		tw = TEXTW(stext) - lrpad + 2; /* 2px right padding */
-		drw_text(drw, m->ww - tw, 0, tw, bh, 0, stext, 0);
+		while (1) {
+			if ((unsigned int)*ts > LENGTH(colors)) { ts++; continue ; }
+			ctmp = *ts;
+			*ts = '\0';
+			drw_text(drw, m->ww - tw + tx, 0, tw - tx, bh, 0, tp, 0);
+			tx += TEXTW(tp) -lrpad;
+			if (ctmp == '\0') { break; }
+			drw_setscheme(drw, scheme[(unsigned int)(ctmp-1)]);
+			*ts = ctmp;
+			tp = ++ts;
+		}
 	}
 
 	for (c = m->clients; c; c = c->next) {
