From 9a292329b6a3a5b23cce29923e59092bd0698a5a Mon Sep 17 00:00:00 2001
From: Ukang'a Dickson <ukanga@users.noreply.github.com>
Date: Tue, 9 Jun 2020 23:17:47 +0300
Subject: [PATCH] Update patch

---
 config.def.h | 23 +++++++++++++++++------
 dwm.c        | 18 ++++++++++++------
 2 files changed, 29 insertions(+), 12 deletions(-)

diff --git a/config.def.h b/config.def.h
index bdaae11..dc20c87 100644
--- a/config.def.h
+++ b/config.def.h
@@ -11,8 +11,8 @@ static const unsigned int gappov    = 10;       /* vert outer gap between window
 static const int smartgaps          = 0;        /* 1 means no outer gap when there is only one window */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
+static const char *fonts[]          = { "FiraCode Nerd Font Mono:style=regular:size=12" };
+static const char dmenufont[]       = "FiraCode Nerd Font Mono:style=regular:size=12";
 static char normbgcolor[]           = "#222222";
 static char normbordercolor[]       = "#444444";
 static char normfgcolor[]           = "#bbbbbb";
@@ -43,7 +43,7 @@ static const unsigned int alphas[][3]      = {
 };
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = { "", "", "", "", "", "", "", "", "" };
 
 static const Rule rules[] = {
 	/* xprop(1):
@@ -55,6 +55,8 @@ static const Rule rules[] = {
 	{ "Firefox", NULL,     NULL,           1 << 8,    0,          0,          -1,        -1 },
 	{ "st",      NULL,     NULL,           0,         0,          1,          -1,        -1 },
 	{ NULL,      NULL,     "Event Tester", 0,         1,          0,           1,        -1 }, /* xev */
+	{ "Slack",   NULL,     NULL,           1 << 3,    0,          0,          -1,        -1 },
+	{ "zoom",    NULL,     NULL,           1 << 4,    0,          0,          -1,        -1 },
 };
 
 /* layout(s) */
@@ -89,11 +91,20 @@ static const Layout layouts[] = {
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", normbgcolor, "-nf", normfgcolor, "-sb", selbordercolor, "-sf", selfgcolor, NULL };
 static const char *termcmd[]  = { "st", NULL };
+static const char *termcmd2[]  = { "st", "-e", "tmux-shell", NULL };
+static const char *raisevolume[] = { "pulsemixer", "--change-volume", "+5", NULL };
+static const char *mutevolume[] = { "pulsemixer", "--toggle-mute", NULL };
+static const char *lowervolume[] = { "pulsemixer", "--change-volume", "-5", NULL };
+#include <X11/XF86keysym.h>
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
+	{ MODKEY,                       0x005c,    spawn,          {.v = termcmd2 } },
+	{ 0,                            XF86XK_AudioLowerVolume, spawn, {.v = lowervolume } },
+	{ 0,                            XF86XK_AudioMute, spawn,   {.v = mutevolume } },
+	{ 0,                            XF86XK_AudioRaiseVolume, spawn, {.v =raisevolume } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
@@ -110,7 +121,7 @@ static Key keys[] = {
 	{ MODKEY|Mod4Mask,              XK_0,      togglegaps,     {0} },
 	{ MODKEY|Mod4Mask|ShiftMask,    XK_0,      defaultgaps,    {0} },
 	{ MODKEY,                       XK_y,      incrihgaps,     {.i = +1 } },
-	{ MODKEY,                       XK_o,      incrihgaps,     {.i = -1 } },
+	{ MODKEY,                       XK_w,      incrihgaps,     {.i = -1 } },
 	{ MODKEY|ControlMask,           XK_y,      incrivgaps,     {.i = +1 } },
 	{ MODKEY|ControlMask,           XK_o,      incrivgaps,     {.i = -1 } },
 	{ MODKEY|Mod4Mask,              XK_y,      incrohgaps,     {.i = +1 } },
@@ -119,7 +130,7 @@ static Key keys[] = {
 	{ MODKEY|ShiftMask,             XK_o,      incrovgaps,     {.i = -1 } },
 	{ MODKEY,                       XK_Return, zoom,           {0} },
 	{ MODKEY,                       XK_Tab,    view,           {0} },
-	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
+	{ MODKEY,                       XK_q,      killclient,     {0} },
 	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
 	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
 	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
@@ -146,7 +157,7 @@ static Key keys[] = {
 	TAGKEYS(                        XK_7,                      6)
 	TAGKEYS(                        XK_8,                      7)
 	TAGKEYS(                        XK_9,                      8)
-	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	{ MODKEY|ShiftMask,             XK_c,      quit,           {0} },
 };
 
 /* button definitions */
diff --git a/dwm.c b/dwm.c
index e4a9313..ded8071 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1269,12 +1269,18 @@ loadxrdb()
       xrdb = XrmGetStringDatabase(resm);
 
       if (xrdb != NULL) {
-        XRDB_LOAD_COLOR("dwm.normbordercolor", normbordercolor);
-        XRDB_LOAD_COLOR("dwm.normbgcolor", normbgcolor);
-        XRDB_LOAD_COLOR("dwm.normfgcolor", normfgcolor);
-        XRDB_LOAD_COLOR("dwm.selbordercolor", selbordercolor);
-        XRDB_LOAD_COLOR("dwm.selbgcolor", selbgcolor);
-        XRDB_LOAD_COLOR("dwm.selfgcolor", selfgcolor);
+        XRDB_LOAD_COLOR("dwm.color7", normfgcolor);
+        XRDB_LOAD_COLOR("dwm.color0", normbgcolor);
+        XRDB_LOAD_COLOR("dwm.color8", normbordercolor);
+        XRDB_LOAD_COLOR("dwm.color7", selfgcolor);
+        XRDB_LOAD_COLOR("dwm.color2", selbgcolor);
+        XRDB_LOAD_COLOR("dwm.color7", selbordercolor);
+        XRDB_LOAD_COLOR("dwm.color8", warnfgcolor);
+        XRDB_LOAD_COLOR("dwm.color3", warnbgcolor);
+        XRDB_LOAD_COLOR("dwm.color1", warnbordercolor);
+        XRDB_LOAD_COLOR("dwm.color7", urgentfgcolor);
+        XRDB_LOAD_COLOR("dwm.color1", urgentbgcolor);
+        XRDB_LOAD_COLOR("dwm.color9", urgentbordercolor);
       }
     }
   }
-- 
2.27.0

